app/controllers/admin/merchants_controller.rb

```diff
@@ -6,4 +6,19 @@ def index
   def show
     @merchant = Merchant.find(params[:id])
   end
+
+  def edit
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def update
+    @merchant = Merchant.find(params[:id])
+    @merchant.update(admin_merchant_params)
+    redirect_to admin_merchant_path(@merchant)
+  end
+
+  private
+  def admin_merchant_params
```

Comment:
solid strong params we should go back and refactor to use them in all of our controllers for sure! great job!